packages/react-ui/src/widgets/CategoryWidgetUI.js

```diff
@@ -300,6 +300,12 @@ function CategoryWidgetUI(props) {
     [maxValue]
   );
 
+  useEffect(() => {
+    if (selectedCategories.length === 0) {
```

Comment:
Ok, so we somehow assume that changing data works well 'removing' the selected categories...

Reply:
In Carto 2, when you lock the category widget you cannot select or unselect categories but in c4r you can select and unselect categories although the widget is blocked. I think that it is a bug and we should fix it in this PR.
If we fix it this new `useEffect` works really well, we can talk about this

Reply:
In Carto 2, when you lock the category widget you cannot select or unselect categories but in c4r you can select and unselect categories although the widget is blocked. I think that it is a bug and we should fix it in this PR.
If we fix it this new useEffect works really well, we can talk about this